raycaster/game/player.py

```diff
@@ -74,14 +74,23 @@ def in_fov(self, angle: float) -> bool:
         return angle_deg >= fov_start or angle_deg <= fov_end
 
     def check_wall_collision(self, dx: int, dy: int):
+        margin = 5  # margin in pixels
+
+        # Check collision with horizontal walls
         if not self.map.is_wall(
-            int((self.x + dx) / self.settings.CELL_SIZE),
+            int(
+                (self.x + dx + margin * math.copysign(1, dx)) / self.settings.CELL_SIZE
```

Comment:
```suggestion
                (self.x + dx + math.copysign(1, dx)) / self.settings.CELL_SIZE
```
we could leave it as so, same below, from I have checked it works fine without the margin